pkg/ingester/ingester.go

```diff
@@ -307,6 +312,12 @@ func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
 		return httpgrpc.Errorf(http.StatusBadRequest, "sample with timestamp %v is older than the maximum accepted age", sample.Timestamp)
 	}
 
+	numLabelNames := len(sample.Metric)
+	if numLabelNames > i.cfg.MaxLabelNamesPerSeries {
+		discardedSamples.WithLabelValues(maxLabelNamesPerSeries).Inc()
+		return httpgrpc.Errorf(http.StatusBadRequest, "sample has %d label names which is greater than the accepted maximum of %d", numLabelNames, i.cfg.MaxLabelNamesPerSeries)
+	}
```

Comment:
Wonder if this should be rolled into `ValidateSample`, and clash with #773?

Reply:
It will make for some merge conflict negotiation, but I do like the idea of validating a sample being done in the `ValidateSample` function